src/client/sandbox/event/message.js

```diff
@@ -55,21 +71,27 @@ export default class MessageSandbox extends SandboxBase {
     }
 
     _onWindowMessage (e, originListener) {
-        const data = typeof e.data === 'string' ? parseJSON(e.data) : e.data;
+        const isOverriddenEventObject = e[IS_OVERRIDDEN_EVENT_OBJECT];
```

Comment:
Drop `isOverriddenEventObject` and rename internal property to `HAS_OVERRIDEN_DATA_PROP`